modules/402-ingress-nginx/images/proxy-failover-iptables/main.go

```diff
@@ -0,0 +1,192 @@
+	saveMarkRule         = strings.Fields("-j CONNMARK --save-mark")
+	restoreHttpMarkRule  = strings.Fields("-p tcp --dport 80 -j CONNMARK --restore-mark")
+	restoreHttpsMarkRule = strings.Fields("-p tcp --dport 443 -j CONNMARK --restore-mark")
+	dnatHttpRule         = strings.Fields("-p tcp --dport 80 -j DNAT --to-destination 169.254.20.11:81")
+	dnatHttpsRule        = strings.Fields("-p tcp --dport 443 -j DNAT --to-destination 169.254.20.11:444")
+	inputAcceptRule      = strings.Fields("-p tcp -m multiport --dport 81,444 -d 169.254.20.11 -m comment --comment ingress-failover -j ACCEPT")
+
+	linkName = "ingressfailover"
+)
+
+func main() {
+	iptablesMgr, err := iptables.NewWithProtocol(iptables.ProtocolIPv4)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = addLinkAndAddress()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// during the failover rollout remove failover-jump-rule setting all traffic to primary
+	_ = iptablesMgr.DeleteIfExists("nat", "PREROUTING", jumpRule...)
+	// do nothing on error, since ingress-failover chain may not exist yet
+
+	// check 81/444 ports accepted
+	err = insertUnique(iptablesMgr, "filter", "INPUT", inputAcceptRule, 1)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// restore conn mark
+	err = insertUnique(iptablesMgr, "mangle", "PREROUTING", restoreHttpMarkRule, 1)
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = insertUnique(iptablesMgr, "mangle", "PREROUTING", restoreHttpsMarkRule, 2)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// fill ingress-failover chain
+	if exists, err := iptablesMgr.ChainExists("nat", chainName); err != nil {
+		log.Fatal(err)
+	} else if !exists {
+		err = iptablesMgr.NewChain("nat", chainName)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+	err = iptablesMgr.Insert("nat", chainName, 1, socketExistsRule...)
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = iptablesMgr.Insert("nat", chainName, 2, markHttpRule...)
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = iptablesMgr.Insert("nat", chainName, 3, markHttpsRule...)
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = iptablesMgr.Insert("nat", chainName, 4, saveMarkRule...)
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = iptablesMgr.Insert("nat", chainName, 5, dnatHttpRule...)
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = iptablesMgr.Insert("nat", chainName, 6, dnatHttpsRule...)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// restore jump rule
+	err = insertUnique(iptablesMgr, "nat", "PREROUTING", jumpRule, 1)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	ticker := time.NewTicker(15 * time.Second)
+	defer ticker.Stop()
+	done := make(chan bool)
+	for {
+		select {
+		case <-done:
+			return
+		case <-ticker.C:
+			err := loop(iptablesMgr)
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+	}
+}
+
+func loop(iptablesMgr *iptables.IPTables) error {
+	resp, err := http.Get("http://127.0.0.1:10254/healthz")
+	defer func(Body io.ReadCloser) {
+		_ = Body.Close()
+	}(resp.Body)
```

Comment:
```suggestion
	defer resp.Body.Close()
```

Reply:
it is go lint fix to avoid unprocessed error warning